index.js

```diff
@@ -173,6 +179,135 @@ app.get("/continentFilter", function (req, res, next) {
+        res.send({cart: []});
+    } else {
+        cartModel.add(cartId, itemId, quantity, imageUrl, _);
+        res.send({cart: cartModel.get(cartId)});
+    }
+});
+
+
+app.put('/cart', function(req, res) {
+    var cartId = req.body.cartId;
+    var itemId = req.body.itemId;
+    var quantity = req.body.quantity;
+
+    if(quantity >= 0) {
+       cartModel.put(cartId, itemId, quantity);
+       sendCartWithSubTotal(cartId, res);
+    } else {
+       res.send('Quantity is a negative.');
+    }
+});
+
+app.delete('/cart', function(req, res) {
+    var cartId = req.query.cartId;
+    var itemId = req.query.itemId;
+
+    if(cartModel.delete(cartId, itemId)) {
+
+    sendCartWithSubTotal(cartId, res);
+
+    } else {
+        res.status(404).send({error: 'Item not found'});
+    };
+});
+
+function sendCartWithSubTotal(cartId, res) {
+
+    var userCart = cartModel.cartById(cartId);
+
+    var subTotal = 0;
+
+    if(userCart.length !== 0) {
+
+        var url = "http://localhost:5000/items?$filter=_id eq '" + userCart[0].itemId + "'";
+
+        for (var i = 1; i < userCart.length; i++) {
+            url += " or _id eq '" + userCart[i].itemId + "'";
+        }
+
+        var requests = [function (callback) {
+            request(url, function (err, response, body) {
+                // JSON body
+                if (err) {
+                    console.log(err);
+                    callback(true);
+                    return;
+                }
+
+                obj = JSON.parse(body);
+                callback(false, obj);
+            });
+        }];
+
+        var resultArray = [];
+
+        async.parallel(requests, function (err, results) {
+            if (err) {
+                console.log(err);
+                res.send(500, "Server Error");
+                return;
+            }
+
+            var query = results[0].value;
+
+
+            for (var i = 0; i < userCart.length; i++) {
+                for (var j = 0; j < query.length; j++) {
+                    if (userCart[i].itemId === query[j]._id) {
+                        resultArray.push({
+                            item: query[i],
+                            quantity: userCart[i].quantity,
+                            imageUrl: userCart[i].imageUrl
+                        });
+                        subTotal += userCart[i].quantity * query[j].price;
+                        break;
+                    }
+                }
+            }
+
+            res.send({cart: resultArray, subTotal: subTotal});
+        });
+    } else {
+        res.send({cart: [], subTotal: 0});
+    }
+}
+
+app.get('/cart', function(req, res) {
+    var cartId = req.query.cartId;
+
+    if(cartId === 'undefined'){
+        var uniqueId = _.uniqueId('cartId-');
+        cartModel.sessions.push({cart: [], cartId: uniqueId });
+        res.send({cart: [], cartId: uniqueId, subTotal: 0});
+    } else {
+
+        var userCart = cartModel.cartById(cartId);
+
+        if(userCart !== null){
```

Comment:
Not big diffrence, but the code is readable if u check for positive things first